core/controllers/cron.py

```diff
@@ -230,6 +231,28 @@ def get(self):
             logging.warning('Deletion jobs for JobModels entities kicked off.')
 
 
+class CronModelsCleanupHandler(base.BaseHandler):
+    """Handler for cleaning up models that are marked as deleted."""
+
+    EIGHT_WEEKS = datetime.timedelta(weeks=8)
+
+    @acl_decorators.can_perform_cron_tasks
+    def get(self):
+        """Cron handler that every week hard deletes all models that are marked
+        as deleted (have deleted field set to True) and were last_updated more
+        than eight months ago.
+        """
+        date_eight_weeks_ago = (
+            datetime.datetime.utcnow() - CronModelsCleanupHandler.EIGHT_WEEKS)
+        for model_class in models.Registry.get_all_storage_model_classes():
+            models_ = model_class.query(model_class.deleted == True).fetch()
```

Comment:
Call this deleted_models, we've never used the trailing underscore convention in python before.